### ISSUE TITLE
Compliance/Earned credits - internal staff can view number of credits requested

### ISSUE THREAD

**Issue by NicoleGovvy** [bcgov/cas-compliance#101]
Partially blocked by: #117 

Description:

CAS staff can review details .

[Figma design](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=21483-5429&t=cm14EZT1DPU56otH-1)

Acceptance Criteria:

As a CAS staff member, given that an operator has requested an EC issuance, when they click 'Review EC request' on the compliance obligation grid, then they first view 'Review [Year] Compliance Summary' with information on the number of credits requested.

- [ ] From [Year] Report heading
- [ ] Emissions attributable for compliance
- [ ] Emissions limit
- [ ] Excess emissions

- [ ] Earned credits heading
- [ ] Earned credits: [number of credits earned]
- [ ] Status of issuance

Earned Credits heading

#### Development Checklist:

- [ ] Update the `auth=authorize("approved_industry_user")` to include CAS staff roles like `auth=authorize(["approved_industry_user", "cas_admin", "cas_analyst"])`
- [ ] Add role detection in `ComplianceSummaryReviewPage.tsx`: `const role = await getSessionRole(); const isCasStaff = role.startsWith("cas_");`
- [ ] Update props interface in `RequestIssuanceReviewComponent.tsx` to include `isCasStaff` prop
- [ ] Update props interface in `RequestIssuanceReviewContent.tsx` to include `isCasStaff` prop
- [ ] Update props interface in `EarnedCredits.tsx` to include `isCasStaff` prop
- [ ] Modify `EarnedCredits.tsx` to conditionally display different content based on the `isCasStaff` prop
- [ ] Create unit tests for the updated components:
  - [ ] Test the `EarnedCredits` component with both `isCasStaff={true}` and `isCasStaff={false}`
  - [ ] Test the conditional rendering logic based on the `isCasStaff` prop
  - [ ] Test the prop passing through the component hierarchy
- [ ] Update the existing test in `bc_obps/compliance/tests/api/test_compliance_summary_issuance.py` to verify that both industry users and CAS staff can access the endpointomments to the code explaining the conditional logic for different user roles

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By/Blocking

Blocked by #62 
Blocking #63 

#### Notes:

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EYpPzWu0h7ZAiP9AQjZ2leMBvcJXiPOHmxXEU6sp8zpYHA?e=UqVryX&nav=eyJoIjoiNDMyODcwNjAzIn0).

I can verify that upon clicking `Request Issuance of Credits`, I can see:
- [x] From [Year] Report heading
- [x] Emissions attributable for compliance
- [x] Emissions limit
- [x] Excess emissions
- [x] Earned credits heading
- [x] Earned credits: [number of credits earned]
- [x] Status of issuance

Moving to Pending PO for review of test. Thank you!

**Comment by fviduya** [comment]
Additional review: We found that the action in the grid should be `Review Credits Issuance Request` based on this [wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=21296-1060&t=JAUHBh7qZk7uUNhu-0). This can be tackled on #63
